src/or/proto_socks.c

```diff
@@ -540,6 +544,15 @@ parse_socks(const char *data, size_t datalen, socks_request_t *req,
     return 0;
   }
 
+  socksver = *data;
```

Comment:
In this case, we _should_ need to cast to `uint8_t *` first.  Or better, we could use the get_uint8() function.

Reply:
Fixed in d71766f462eb2630087f51a474a5f28a09f0f949.